service/src/main.rs

```diff
@@ -162,157 +184,189 @@ impl Inclusion for Arc<InclusionService> {
 }
 
 impl InclusionService {
-    async fn job_worker(&self, mut job_receiver: mpsc::UnboundedReceiver<Job>) {
-        info!("Job worker started");
+    /// A worker that recives [Job]s by a channel and drives them to completion
+    /// Each state change is handled for [JobStatus] that creates an atomic unit of
+    /// work to be completed async. Once completed, work is commetted into
+    /// a queue data base that can be recovered to take up where a job was left off.
+    ///
+    /// Once the job comes to an ending successful or failed state,
+    /// the job is atomically removed from the queue and added to a results data base.
+    async fn job_worker(
+        &self,
+        mut job_receiver: mpsc::UnboundedReceiver<Job>,
+    ) -> Result<Option<JobStatus>, InclusionServiceError> {
+        debug!("Job worker started");
         while let Some(job) = job_receiver.recv().await {
             debug!(
                 "job worker received job for commitment: {}",
                 hex::encode(job.commitment.clone())
             );
-            let client = Arc::clone(&self.client);
-            tokio::spawn(prove(
-                job,
-                client,
-                self.queue_db.clone(),
-                self.proof_db.clone(),
-            ));
```

Comment:
Unclear if we need to spawn what replaces this :thinking:

Reply:
it doesn't replace job_worker, which [runs in perpetuity](https://github.com/celestiaorg/eq-service/blob/main/service/src/main.rs#L350). We use `task::spawn` here so it can immediately start waiting for the next job obtained from a user via the [job_receiver](https://github.com/celestiaorg/eq-service/blob/main/service/src/main.rs#L167)